core/tee/tee_svc_storage.c

```diff
@@ -297,6 +297,54 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 	return res;
 }
 
+static TEE_Result tee_svc_storage_update_head(struct tee_obj *o)
+{
+	TEE_Result res = TEE_SUCCESS;
+	size_t bytes;
+	struct tee_svc_storage_head head;
+	const struct tee_file_operations *fops;
+	int32_t old_off;
+
+	fops = o->pobj->fops;
+
+	/* save original offset */
+	res = fops->seek(o->fh, 0, TEE_DATA_SEEK_CUR, &old_off);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = fops->seek(o->fh, 0, TEE_DATA_SEEK_SET, NULL);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	/* read head */
+	bytes = sizeof(struct tee_svc_storage_head);
+	res = fops->read(o->fh, &head, &bytes);
```

Comment:
You could skip the read step if you'd only update the position field in the file.